src/app/public/modules/data-manager/data-manager-ag-grid-adapter.directive.ts

```diff
@@ -0,0 +1,170 @@
+import {
+  AfterContentInit,
+  Directive,
+  ChangeDetectorRef,
+  ContentChildren,
+  QueryList,
+  Input
+} from '@angular/core';
+
+import {
+  Subscription
+} from 'rxjs';
+
+import {
+  AgGridAngular
+} from 'ag-grid-angular';
+
+import {
+  ColumnMovedEvent
+} from 'ag-grid-community';
+
+import {
+  SkyDataManagerService
+} from './data-manager.service';
+
+import {
+  SkyDataViewConfig,
+  SkyDataManagerState,
+  SkyDataManagerSortOption
+} from './models';
+
+@Directive({
+  selector: '[skyAgGridDataManager]'
+})
+export class SkyAgGridDataManagerAdapterDirective implements AfterContentInit {
+
+  @Input()
+  public viewConfig: SkyDataViewConfig;
+
+  private currentAgGrid: AgGridAngular;
+
+  @ContentChildren(AgGridAngular, { descendants: true })
+  public agGridList: QueryList<AgGridAngular>;
+
+  private dataStateSub: Subscription;
+
+  constructor(
+    private changeDetector: ChangeDetectorRef,
+    private dataManagerSvc: SkyDataManagerService) { }
+
+  public ngAfterContentInit() {
+    this.checkForAgGrid();
+
+    this.agGridList.changes.subscribe(() => this.checkForAgGrid());
+  }
+
+  private checkForAgGrid() {
+    const agGridCount = this.agGridList.length;
+
+    if (agGridCount > 1) {
+      throw new Error(
+        'A data view may only have one active ag-Grid child component instance at a time.'
+      );
+    } else if (agGridCount === 0) {
+      this.unregisterAgGrid();
+    } else if (this.agGridList.first !== this.currentAgGrid) {
+      this.registerAgGrid();
+    }
+  }
+
+  private unregisterAgGrid() {
+    this.currentAgGrid = undefined;
+
+    if (this.dataStateSub) {
+      this.dataStateSub.unsubscribe();
+    }
+  }
+
+  private registerAgGrid() {
+    this.unregisterAgGrid();
+
+    const agGrid = this.agGridList.first;
```

Comment:
`this.agGridList.first` is used a few times in here. Should we encapsulate that?